typescript/raven.d.ts

```diff
@@ -0,0 +1,195 @@
+    maxMessageLength?: number;
+
+    /** Override the default HTTP data transport handler. */
+    transport?: (options: RavenTransportOptions) => void;
+}
+
+export interface RavenStatic {
+
+    /** Raven.js version. */
+    VERSION: string;
+
+    Plugins: { [id: string]: RavenPlugin };
+
+    /*
+     * Allow Raven to be configured as soon as it is loaded
+     * It uses a global RavenConfig = {dsn: '...', config: {}}
+     *
+     * @return undefined
+     */
+    afterLoad(): void;
+
+    /*
+     * Allow multiple versions of Raven to be installed.
+     * Strip Raven from the global context and returns the instance.
+     *
+     * @return {Raven}
+     */
+    noConflict(): RavenStatic;
+
+    /*
+     * Configure Raven with a DSN and extra options
+     *
+     * @param {string} dsn The public Sentry DSN
+     * @param {object} options Optional set of of global options [optional]
+     * @return {Raven}
+     */
+    config(dsn: string, options?: RavenOptions): RavenStatic;
+
+    /*
+     * Installs a global window.onerror error handler
+     * to capture and report uncaught exceptions.
+     * At this point, install() is required to be called due
+     * to the way TraceKit is set up.
+     *
+     * @return {Raven}
+     */
+    install(): RavenStatic;
+
+    /*
+     * Adds a plugin to Raven
+     *
+     * @return {Raven}
+     */
+    addPlugin(plugin: RavenPlugin, ...pluginArgs: any[]): RavenStatic;
+
+    /*
+     * Wrap code within a context so Raven can capture errors
+     * reliably across domains that is executed immediately.
+     *
+     * @param {object} options A specific set of options for this context [optional]
+     * @param {function} func The callback to be immediately executed within the context
+     * @param {array} args An array of arguments to be called with the callback [optional]
+     */
+    context(func: Function, ...args: any[]): void;
+    context(options: RavenOptions, func: Function, ...args: any[]): void;
+
+    /*
+     * Wrap code within a context and returns back a new function to be executed
+     *
+     * @param {object} options A specific set of options for this context [optional]
+     * @param {function} func The function to be wrapped in a new context
+     * @return {function} The newly wrapped functions with a context
+     */
+    wrap(func: Function): Function;
+    wrap(options: RavenOptions, func: Function): Function;
+    wrap<T extends Function>(func: T): T;
+    wrap<T extends Function>(options: RavenOptions, func: T): T;
+
+    /*
+     * Uninstalls the global error handler.
+     *
+     * @return {Raven}
+     */
+    uninstall(): RavenStatic;
+
+    /*
+     * Manually capture an exception and send it over to Sentry
+     *
+     * @param {error} ex An exception to be logged
+     * @param {object} options A specific set of options for this error [optional]
+     * @return {Raven}
+     */
+    captureException(ex: Error, options?: RavenOptions): RavenStatic;
```

Comment:
captureException technically allows a string as first argument as well, and we fall back to captureMessage.

BUT this might be a good thing to enforce in the type definition. The fallback was done because things can throw strings and whatnot.